nano/core_test/socket.cpp

```diff
@@ -238,39 +222,20 @@ TEST (socket, max_connections_per_subnetwork)
 {
 	nano::test::system system;
 
-	auto node_flags = nano::inactive_node_flag_defaults ();
-	node_flags.read_only = false;
+	nano::node_flags node_flags;
 	// disabling IP limit because it will be used the same IP address to check they come from the same subnetwork.
 	node_flags.disable_max_peers_per_ip = true;
 	node_flags.disable_max_peers_per_subnetwork = false;
-	nano::inactive_node inactivenode (nano::unique_path (), node_flags);
-	auto node = inactivenode.node;
+	nano::node_config node_config = system.default_config ();
+	node_config.network.max_peers_per_subnetwork = 3;
+	auto node = system.add_node (node_config, node_flags);
 
 	ASSERT_TRUE (node->flags.disable_max_peers_per_ip);
 	ASSERT_FALSE (node->flags.disable_max_peers_per_subnetwork);
 
-	auto server_port = system.get_available_port ();
-	boost::asio::ip::tcp::endpoint listen_endpoint{ boost::asio::ip::address_v6::any (), server_port };
-
-	const auto max_subnetwork_connections = node->network_params.network.max_peers_per_subnetwork;
+	const auto max_subnetwork_connections = node->config.network.max_peers_per_subnetwork;
 	ASSERT_GE (max_subnetwork_connections, 1);
 
-	const auto max_global_connections = 1000;
```

Comment:
Were these 1000 connection limits brought down to 3 mostly because we don't need 1000 connections to test the limiter?

Reply:
I don't think this limit was used anywhere actually, just a dummy number. The interesting limit is the `config.network.max_peers_per_ip`.To test it, this testcase used to spin up 20 nodes (which was the default limit) and previously this * barely * worked. With this PR, with continuously running IO, it was prohibitively expensive to simulate that many nodes on the low-spec github runner, so I needed to make this configurable and limit it to something more reasonable.